src/components/HourSelector/index.tsx

```diff
@@ -0,0 +1,199 @@
+          className={classNames(
+            'absolute',
+            'w-10 h-10 rounded-full shadow-lg bg-white transition',
+            'grid place-items-center hover:bg-gray-800 hover:text-white',
+            'focus:outline-none focus:ring-2 focus:ring-gray-800',
+            'focus:ring-offset-2 focus:ring-offset-white z-50',
+            !hasMobileSize && '-top-4 right-6 transform translate-x-full',
+            hasMobileSize && 'top-1/2 -right-12 transform -translate-y-1/2'
+          )}
+          onClick={() => setIsOpened(false)}
+        >
+          <CrossIcon />
+        </button>
+      )}
+      <div
+        className={classNames(
+          'absolute w-48 h-48 grid place-items-center inset-0'
+        )}
+      >
+        <div
+          style={{ borderWidth: 40 }}
+          className={classNames(
+            'outer-ring w-24 h-24 rounded-full',
+            'border-white box-content opacity-90',
+            'transform transition',
+            isOpened ? 'scale-110 opacity-90 delay-200' : 'scale-50 opacity-0'
+          )}
+        />
+      </div>
+      <div className="absolute w-48 h-48 grid place-items-center inset-0">
+        <button
+          aria-label={!isOpened ? 'open-hour-selector' : undefined}
+          onClick={() => !isOpened && setIsOpened(true)}
+          tabIndex={isOpened ? -1 : 0}
+          className={classNames(
+            'inner-ring w-24 h-24',
+            'flex flex-col place-items-center justify-center',
+            'rounded-full bg-white shadow-lg delay-100',
+            'transform origin-top-right transition',
+            'focus:outline-none',
+            !isOpened && [
+              'focus:ring-3 focus:ring-gray-800',
+              'hover:bg-gray-800 hover:text-white',
+              'select-none cursor-pointer',
+            ]
+          )}
+          style={{
+            transform: classNames(
+              isOpened && 'scale(1) translate(0, 0)',
+              !isOpened &&
+                !hasMobileSize &&
+                'scale(.75) translate(80px, -80px)',
+              !isOpened && hasMobileSize && 'scale(.5) translate(145%, 348px)'
+            ),
+          }}
+        >
+          <span className="text-4xl leading-8">{activeHourKey}</span>
+          <span className="text-sm">Uhr</span>
+        </button>
+      </div>
+      <div
+        className={classNames(
+          'absolute w-48 h-48 grid place-items-center',
+          'inset-0 transform transition',
+          isOpened ? 'scale-100 opacity-90 delay-200' : 'scale-50 opacity-0'
+        )}
+      >
+        <svg
+          width="195"
+          height="195"
+          viewBox="0 0 195 195"
+          fill="none"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          {Object.keys(hoursPositions).map((key) => {
+            const { x, y } = hoursPositions[key]
+            return (
+              <g
+                key={key}
+                role="button"
```

Comment:
Amazing that you made this accessible!

Reply:
Haha. Not too sure about how _correctly_ this is implemented though. Better than nothing I guess... :D If you have some hints, please let me know! :)

Reply:
Exactly, it's good to move towards more accessibility, even if not everything is a 100% correct. I am no expert myself, so consider this just another idea:

If I see it correctly, we select the hour here, so this could receive the role [`option`](https://w3c.github.io/aria/#option). The parent `svg` or `div` would therefore receive the role [`listbox`](https://w3c.github.io/aria/#listbox) to group the `option`s and indicate that a selection can be made here - a selection of different hours.